crates/ra_hir_ty/src/diagnostics.rs

```diff
@@ -242,7 +253,7 @@ impl Diagnostic for MismatchedArgCount {
 impl AstDiagnostic for MismatchedArgCount {
     type AST = ast::CallExpr;
     fn ast(&self, db: &dyn AstDatabase) -> Self::AST {
-        let root = db.parse_or_expand(self.source().file_id).unwrap();
+        let root = db.parse_or_expand(self.file).unwrap();
         let node = self.source().value.to_node(&root);
         ast::CallExpr::cast(node).unwrap()
```

Comment:
This place is a bit suspicious: in other diagnostics, I was able to remove the `.source()` calls to avoid unnecessary `.clone()` and `.unwrap()` calls, but this one stores the `call_expr` as `pub call_expr: AstPtr<ast::Expr>,` instead of `AstPtr<ast::CallExpr>` and it's not possible to use the same trick.